src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -548,7 +549,8 @@ void insert_reorders_in_dir(program& p, const std::map<program_node*, format::ty
         if (!next->is_in_data_flow())
             continue;
 
-        if (fmt_map.count(next) > 0 && fmt_map.at(next) == fmt)
+        if (fmt_map.count(next) > 0 &&
+            (fmt_map.at(next) == fmt || (fmt_map.at(next) != fmt && next->template is_type<gather_nd>())))
```

Comment:
Can remove "fmt_map.at(next) != fmt &&" here.
And expend condition to  gather and gather_elements if required.

Reply:
I agree and applied it.